tracer/src/Datadog.Tracer.Native/cor_profiler.cpp

```diff
@@ -1626,6 +1634,41 @@ HRESULT STDMETHODCALLTYPE CorProfiler::JITCompilationStarted(FunctionID function
             return S_OK;
         }
 
+        // *********************************************************************
+        // Checking if the caller has an explicit static constructor.
+        // If it does, we delay instrumenting this and let the static constructor get instrumented instead.
+        // Bypassing for calls that we explicitly want to instrument (e.g. IIS startup hook)
+        // *********************************************************************
+        if (caller.name == WStr(".cctor"))
+        {
+            // Injecting into the static constructor is generally what we want to do
+            can_skip_startup_hook_callsite = false;
+        }
+
+        if (can_skip_startup_hook_callsite)
+        {
+            mdTypeDef mainTypeDef;
+            hr = metadataImport->FindTypeDefByName(caller.type.name.c_str(), mdTokenNil, &mainTypeDef);
+            if (FAILED(hr) || !mainTypeDef)
```

Comment:
checking hr is enough I think (same for below)

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/6154/commits/3824c79399f87a008b20c652ea99942b172bc709